Sources/PackageLoading/ModuleMapGenerator.swift

```diff
@@ -134,6 +137,17 @@ public struct ModuleMapGenerator {
         }
         diagnoseInvalidUmbrellaHeader(includeDir.appending(component: target.c99name))
 
+        // Otherwise, if 'include' contains only header files and no subdirectories, use it as the umbrella directory (this is case 3 at https://github.com/apple/swift-package-manager/blob/master/Documentation/Usage.md#creating-c-language-targets).
+        if files.count == walked.count {
+            try createModuleMap(inDir: wd, type: .directory(includeDir))
+            return
+        }
+        
+        // Otherwise, the target's public headers are considered to be incompatible with modules.  Other C targets can still import them, but Swift won't be able to see them.  This is documented as an error, but because SwiftPM has previously allowed it (creating module maps that then cause errors when used), we instead emit a warning and for now, continue to emit what SwiftPM has historically emitted (an umbrella directory include).
+        warningStream <<< "warning: the include directory of target '\(target.name)' has "
```

Comment:
This is writing to `stdout` when used by sourcekit-lsp, which breaks us.  I should be able to fix sourcekit-lsp to not let you write to stdout, but this is bad behaviour for a library in general.  It should at least be on stderr.

Reply:
I totally agree.  I found that this file was doing this already, so I was following suit, but I am preparing the PR to switch to DiagnosticsEngine as a separate PR.  I didn't realize that this change would cause breakage though (I guess any of the existing uses of `warningStream` would as well — I thought they were pointed at `stderr` and not `stdout`).